src/Compilers/Core/Portable/Collections/ImmutableArrayExtensions.cs

```diff
@@ -539,5 +539,38 @@ public static int Count<T>(this ImmutableArray<T> items, Func<T, bool> predicate
 
             return dictionary;
         }
+
+        /// <summary>
+        /// Checks whether two arrays of elements who are IEquatable are equal.
+        /// </summary>
+        internal static bool SequenceEquals<T>(this ImmutableArray<T> one, ImmutableArray<T> other) where T : IEquatable<T>
+        {
+            bool oneIsDefault = one.IsDefault;
+            if (oneIsDefault != other.IsDefault)
+            {
+                return false;
+            }
+
+            if (oneIsDefault)
+            {
+                return true;
+            }
+
+            int length = one.Length;
+            if (length != other.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < length; i++)
+            {
+                if (!one[i].Equals(other[i]))
```

Comment:
What if this is ImmutableArray of a reference type?

Reply:
I didn't understand the question. The type in question must be `IEquatable` and we will use its notion of equality (this `Equals` is `IEquatable<T>.Equals(T other)`).

Reply:
The problem here is htat one[i] may be null.  And you could null ref.  Better would be to use ```EqualityComparer<T>.Default.Equals(...)``` so that it works properly for nulls as well.